doc/modules/calibration.rst

```diff
@@ -111,30 +123,63 @@ predicted probabilities of the `k` estimators in the
-parameter. `'sigmoid'` corresponds to a parametric approach based on Platt's
-logistic model [3]_, i.e. :math:`p(y_i = 1 | f_i)` is modeled as
-:math:`\sigma(A f_i + B)` where :math:`\sigma` is the logistic function, and
-:math:`A` and :math:`B` are real numbers to be determined when fitting the
-regressor via maximum likelihood. `'isotonic'` will instead fit a
-non-parametric isotonic regressor, which outputs a step-wise non-decreasing
-function (see :mod:`sklearn.isotonic`).
-
-An already fitted classifier can be calibrated by setting `cv="prefit"`. In
-this case, the data is only used to fit the regressor. It is up to the user
+Alternatively an already fitted classifier can be calibrated by setting
+`cv="prefit"`. In this case, the data is not split and all of it is used to
+fit the regressor. It is up to the user
 make sure that the data used for fitting the classifier is disjoint from the
 data used for fitting the regressor.
 
-:class:`CalibratedClassifierCV` can calibrate probabilities in a multiclass
-setting if the base estimator supports multiclass predictions. The classifier
-is calibrated first for each class separately in a one-vs-rest fashion [4]_.
-When predicting probabilities, the calibrated probabilities for each class
+:class:`CalibratedClassifierCV` supports the use of two 'calibration'
+regressors; 'sigmoid' and 'isotonic'. Both these regressors only
+support 1-dimensional data (e.g., binary classification output) but are
+extended for multiclass classification if the `base_estimator` supports
+multiclass predictions. For multiclass predictions,
+:class:`CalibratedClassifierCV` calibrates for
+each class separately in a :ref:`ovr_classification` fashion [4]_. When
+predicting
+probabilities, the calibrated probabilities for each class
 are predicted separately. As those probabilities do not necessarily sum to
 one, a postprocessing is performed to normalize them.
 
 The :func:`sklearn.metrics.brier_score_loss` may be used to evaluate how
 well a classifier is calibrated.
 
+Sigmoid
+^^^^^^^
+
+The sigmoid regressor is based on Platt's logistic model [3]_:
+
+.. math::
+       p(y_i = 1 | f_i) = \frac{1}{1 + \exp(A f_i + B)}
+
+where :math:`y_i` is the true label of sample :math:`i` and :math:`f_i`
+is output of the classifier for sample :math:`i`. :math:`A` and :math:`B`
+are real numbers to be determined when fitting the regressor via maximum
+likelihood.
+
+The sigmoid method is biased in that it assumes the :ref:`calibration curve
+<calibration_curve>` of the un-calibrated model has a sigmoid shape [1]_. It
+is thus most effective when the un-calibrated model is over-confident.
+
+Isotonic
+^^^^^^^^
+
+The 'isotonic' method fits a non-parametric isotonic regressor, which outputs
+a step-wise non-decreasing function (see :mod:`sklearn.isotonic`). It
+minimizes:
+
+.. math::
+       \sum_i (y_i - f_i)^2
+
+subject to :math:`\f_i \le f_j`. This method is more general when compared to
```

Comment:
The constraint is y_i < y_j whenever f_i < f_j

Though we should not use y_i: y_i is used before and corresponds to the true target (0 or 1).

Reply:
I was confused at this part but I think y_i here does mean the true target.

![image](https://user-images.githubusercontent.com/23182829/85780604-17408a00-b725-11ea-860b-286b3c71bffd.png)

from: https://www.ncbi.nlm.nih.gov/pmc/articles/PMC4180410/
(sorry for funny screenshot)

Reply:
"subject to :math:`\f_i \le f_j`" is ambiguous as `j` is not defined. I think you should reproduce the full formula you quoted above only using `f_i` / `f_{i+1}` instead of the `\tilde{s}_i` / `\tilde{s}_{i+1}` notation.